apps/aecore/src/aec_dev_reward.erl

```diff
@@ -0,0 +1,122 @@
+                false ->
+                    lager:error("sum of shares for protocol beneficiaries is too large (> ~p)",
+                                [?TOTAL_SHARES]),
+                    exit({invalid_protocol_beneficiaries, sum_shares_too_large})
+            end;
+        {error, Reason} ->
+            lager:error("invalid protocol beneficiaries: ~p", [Reason]),
+            exit({invalid_protocol_beneficiaries, Reason})
+    end.
+
+cfg(Key, Default) ->
+    aeu_env:user_config([<<"chain">>, Key], Default).
+
+env(Key, Default) ->
+    aeu_env:get_env(aecore, Key, Default).
+
+enabled() ->
+    env(dev_reward_enabled, ?ENABLED).
+beneficiaries() ->
+    env(dev_reward_beneficiaries, ?BENEFICIARIES).
+total_shares() ->
+    ?TOTAL_SHARES.
+allocated_shares() ->
+    case env(dev_reward_allocated_shares, undefined) of
+        undefined ->
+            AllocShares = lists:foldl(fun ({_, X}, A) -> A + X end, 0, beneficiaries()),
+            application:set_env(aecore, dev_reward_allocated_shares, AllocShares),
+            AllocShares;
+        Res ->
+            Res
+    end.
+
+activated(Height) ->
+    case env(dev_reward_activated, undefined) of %% for eunit to avoid mocking
+        undefined ->
+            aec_hard_forks:protocol_effective_at_height(Height) >= ?FORTUNA_PROTOCOL_VSN;
+        Val when is_boolean(Val) ->
+            Val
+    end.
+
+
+unparse_beneficiary({PK, Share}) ->
+    <<(aeser_api_encoder:encode(account_pubkey, PK))/binary, ":", (integer_to_binary(Share))/binary>>.
+
+parse_beneficiary(BeneficiaryShareStr) ->
+    Regex = "^(?'account'ak_[1-9A-HJ-NP-Za-km-z]*):(?'share'[0-9]+)$",
+    case re:split(BeneficiaryShareStr, Regex, [trim, group, {return, binary}]) of
+        [[<<>>, Account, Share]] ->
+            case aeser_api_encoder:safe_decode(account_pubkey, Account) of
+                {ok, PubKey} ->
+                    {PubKey, binary_to_integer(Share)};
+                {error, _} ->
+                    error({invalid_protocol_beneficiary, Account})
+            end;
+        _ ->
+            error({invalid_protocol_beneficiary_share, BeneficiaryShareStr})
+    end.
+
+parse_beneficiaries(BeneficiarySharesStrs) ->
+    try lists:foldl(fun (Str, Acc) -> [parse_beneficiary(Str) | Acc] end,
+                    [], BeneficiarySharesStrs) of
+        Parsed ->
+            {ok, Parsed}
+    catch
+        error:Reason ->
+            {error, Reason}
+    end.
+
+
+split(BeneficiaryReward1, BeneficiaryReward2, NewestNodeHeight) ->
+    case {activated(NewestNodeHeight), enabled()} of
+        {true, true} ->
+            AllocShares = allocated_shares(),
+            TotalShares = total_shares(),
+            AbsContrib1 = BeneficiaryReward1 * AllocShares div TotalShares,
+            AbsContrib2 = BeneficiaryReward2 * AllocShares div TotalShares,
+            DevContrib  = AbsContrib1 + AbsContrib2,
+            {_, DevRewards} =
```

Comment:
What prevents value to be matched with 0?
```suggestion
            {0, DevRewards} =
```
If value is not zero, it means that either some coins are not-assigned-and-not-locked, or that some tokens are created, right?

Reply:
Unresolving as questions seems still applicable to me.

Reply:
This isn't really easy question to answer - it's more in the math area. 
My suspicion is that there *is* a combination of shares which could result in accumulation of not-assigned-and-not-locked amount at the end of fold, as normal float division could end up with result like 13.55, but with DIV this is cut to 13. 
So I think the best idea here is to distribute the amounts as is, but if we end up with some not assigned amount, take it and add add it to the last beneficiary - it would be just small remains (if any) - will implement this.